app/Services/Model/Imp/SummitOrderService.php

```diff
@@ -594,8 +594,8 @@ public function run(array $formerState): array
 
                 Log::debug(sprintf("adding %s usage to promo code %s", $qty, $promo_code->getId()));
 
-                $this->lock_service->lock('promocode.' . $promo_code->getId() . '.usage.lock', function () use ($promo_code, $qty) {
-                    $promo_code->addUsage($qty);
+                $this->lock_service->lock('promocode.' . $promo_code->getId() . '.usage.lock', function () use ($promo_code, $qty, $owner_email) {
```

Comment:
we missed to update the compensatory transaction here 
https://github.com/OpenStackweb/summit-api/blob/f084b54f065992303c49f53b5ba23d022b4cacc0/app/Services/Model/Imp/SummitOrderService.php#L620
we need to update the signature of removeUsage here
https://github.com/OpenStackweb/summit-api/blob/f084b54f065992303c49f53b5ba23d022b4cacc0/app/Models/Foundation/Summit/Registration/PromoCodes/SummitRegistrationPromoCode.php#L363
to accept an optional owner_email address and override the proper logic on the speakers promo code trait

Reply:
also we need to review all the uses of removeUsage ( whole project)